elide-core/src/main/java/com/yahoo/elide/core/filter/FilterPredicate.java

```diff
@@ -7,108 +7,109 @@
 
 import com.yahoo.elide.core.EntityDictionary;
 import com.yahoo.elide.core.Path;
+import com.yahoo.elide.core.Path.PathElement;
 import com.yahoo.elide.core.RelationshipType;
 import com.yahoo.elide.core.RequestScope;
 import com.yahoo.elide.core.exceptions.InvalidOperatorNegationException;
 import com.yahoo.elide.core.filter.expression.FilterExpression;
 import com.yahoo.elide.core.filter.expression.Visitor;
+import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.NonNull;
 import org.apache.commons.lang3.StringUtils;
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 /**
  * Predicate class.
  */
 @EqualsAndHashCode
 public class FilterPredicate implements FilterExpression, Function<RequestScope, Predicate> {
+    private static final String UNDERSCORE = "_";
+    private static final String PERIOD = ".";
+
     @Getter @NonNull private Path path;
     @Getter @NonNull private Operator operator;
     @Getter @NonNull private List<Object> values;
-    private static final String UNDERSCORE = "_";
-    private static final String PERIOD = ".";
 
     public static boolean toManyInPath(EntityDictionary dictionary, Path path) {
         return path.getPathElements().stream()
                 .map(element -> dictionary.getRelationshipType(element.getType(), element.getFieldName()))
                 .anyMatch(RelationshipType::isToMany);
     }
 
-    public FilterPredicate(Path.PathElement pathElement, Operator op, List<Object> values) {
+    public FilterPredicate(PathElement pathElement, Operator op, List<Object> values) {
         this(new Path(Arrays.asList(pathElement)), op, values);
     }
 
-    public FilterPredicate(Path.PathElement pathElement, Operator op) {
-        this(new Path(Arrays.asList(pathElement)), op, Collections.emptyList());
-    }
-
-    public FilterPredicate(Path path, Operator op) {
-        this(path, op, Collections.emptyList());
+    public FilterPredicate(FilterPredicate copy) {
+        this(copy.path, copy.operator, copy.values);
     }
 
     public FilterPredicate(Path path, Operator op, List<Object> values) {
-        this.path = path;
+        this.path = new Path(path);
         this.operator = op;
-        this.values = values;
-    }
+        this.values = new ArrayList<>(values);
 
-    public FilterPredicate(FilterPredicate copy) {
-        this.path = new Path(new ArrayList<>(copy.getPath().getPathElements()));
-        this.operator = copy.getOperator();
-        this.values = new ArrayList<>(copy.getValues());
+        // TODO: figure out why this cannot be here :(
```

Comment:
Should we resolve this before merging?

Also, what problems are we seeing with this?